src/journal-remote/journal-upload-journal.c

```diff
@@ -300,6 +309,17 @@ static size_t journal_input_callback(void *buf, size_t size, size_t nmemb, void
                           u->entries_sent, u->current_cursor);
         }
 
+        if (filled > 0 && u->compression != COMPRESSION_NONE) {
+                size_t compressed_size;
+                r = compress_blob(u->compression, compression_buffer, filled, buf, size * nmemb, &compressed_size, u->compression_level);
+                if (r < 0) {
+                        log_error_errno(r, "Failed to compress %zu bytes using Compression=%s, CompressionLevel=%d)",
+                                        filled, compression_to_string(u->compression), u->compression_level);
+                        return CURL_READFUNC_ABORT;
+                }
+                return compressed_size;
```

Comment:
hmm, so compress_blob() can fail if the compressed data  is larger than the uncompressed data. You need to handle that somehow.

Reply:
what do you suggest here?

Reply:
maybe iteratively halve the input buffer size, until it fits, and then store the rest in some per-connection buffer, and use that on the next iteration first.